### PR TITLE
Mipmap generation for WebGPU 2D textures

### PR DESCRIPTION
**Pull request by mvaligursky**
- WebgpuMipmapRenderer implementation, which uses render passes to render from top to lower mipmap levels.
- Mipmaps are generated on WebGPU if rendering to a render target with mipmaps, for loaded png/jpg textures, and for color buffer grab pass. Only 2D textures for now, now cube/volume textures.
- moved deprecated Texture.autoMipmap to deprecated.js

### Changea to the texture loading flow:
- **Original flow**: load texture with default settings (mip, filtering ...), then use associated asset meta data to apply override settings.
- **New flow**: extract override settings and pass them to .Open function which creates the texture resource, to allow texture to be created without the need to override later.
Background: WebGPU creates texture immediatelly, instead of on first use, and it would have to handle texture re-creation.
Limitation: WebGPU does not currently allow texture.mipmaps to be changed after the creation. Subject to discussion / change / further work.

### Other changes:
- [Fix] area light LUT texture no longer generates mipmaps (those were unused anywys)

### References:
@toji for WebGPU Textures - best practices: https://toji.dev/webgpu-best-practices/img-textures#generating-mipmaps

### Result - textures no longer shimmer

https://user-images.githubusercontent.com/59932779/234560689-1e406ec0-69a7-455a-aca2-600a6a6bee11.mov

### CODE REVIEW

File: src/framework/parsers/texture/basis.js [hunk 1]
```diff
@@ -45,21 +45,22 @@ class BasisParser {
...
+        textureOptions.name ??= url;
+        textureOptions.width ??= data.width;
+        textureOptions.height ??= data.height;
+        textureOptions.format ??= data.format;
+        textureOptions.cubemap ??= data.cubemap;
+        textureOptions.addressU ??= data.cubemap ? ADDRESS_CLAMP_TO_EDGE : ADDRESS_REPEAT;
+        textureOptions.addressV ??= data.cubemap ? ADDRESS_CLAMP_TO_EDGE : ADDRESS_REPEAT;
+        textureOptions.levels ??= data.levels;
+
+        const texture = new Texture(device, textureOptions);
```
Comment (slimbuck) [inline src/framework/parsers/texture/basis.js:63]: Probably better to do:
```
    const texture = new Texture(device, {
            name: url,
            // #if _PROFILER
            profilerHint: TEXHINT_ASSET,
            // #endif
            addressU: data.cubemap ? ADDRESS_CLAMP_TO_EDGE : ADDRESS_REPEAT,
            addressV: data.cubemap ? ADDRESS_CLAMP_TO_EDGE : ADDRESS_REPEAT,
            width: data.width,
            height: data.height,
            format: data.format,
            cubemap: data.cubemap,
            levels: data.levels,
            ...textureOptions
        });
```

Comment (mvaligursky) [reply src/framework/parsers/texture/basis.js]: yep definitely simpler. Done.

**Review by slimbuck — APPROVED** [review]

File: src/deprecated/deprecated.js [hunk 2]
```diff
@@ -562,6 +562,17 @@ Object.defineProperties(Texture.prototype, {
             Debug.deprecated('pc.Texture#_glTexture is no longer available, use Use pc.Texture.impl._glTexture instead.');
             return this.impl._glTexture;
         }
+    },
+
+    autoMipmap: {
+        get: function () {
+            Debug.deprecated('pc.Texture#autoMipmap is deprecated, use Use pc.Texture#mipmaps instead.');
```
Comment (willeastcott) [inline src/deprecated/deprecated.js:569]:
```suggestion
            Debug.deprecated('pc.Texture#autoMipmap is deprecated, use pc.Texture#mipmaps instead.');
```

File: src/deprecated/deprecated.js [hunk 3]
```diff
@@ -562,6 +562,17 @@ Object.defineProperties(Texture.prototype, {
...
+            return this._mipmaps;
+        },
+        set: function (value) {
+            Debug.deprecated('pc.Texture#autoMipmap is deprecated, use Use pc.Texture#mipmaps instead.');
```
Comment (willeastcott) [inline src/deprecated/deprecated.js:573]:
```suggestion
            Debug.deprecated('pc.Texture#autoMipmap is deprecated, use pc.Texture#mipmaps instead.');
```

File: src/framework/parsers/texture/img.js [hunk 4]
```diff
@@ -69,16 +69,24 @@ class ImgParser {
         }
     }
 
-    open(url, data, device) {
+    open(url, data, device, textureOptions = {}) {
+
+
+        console.log("!!!!!", url);
```
Comment (willeastcott) [inline src/framework/parsers/texture/img.js:75]: Uhm. Remove?

File: src/platform/graphics/texture.js [hunk 5]
```diff
@@ -496,6 +481,11 @@ class Texture {
     set mipmaps(v) {
         if (this._mipmaps !== v) {
             this._mipmaps = v;
+
+            if (this.device.isWebGPU) {
+                Debug.warn("Texture#mipmaps: mipmap property is currently not allowed to be changed on WebGPU, create the texture approproately.", this);
```
Comment (willeastcott) [inline src/platform/graphics/texture.js:486]:
```suggestion
                Debug.warn("Texture#mipmaps: mipmap property is currently not allowed to be changed on WebGPU, create the texture appropriately.", this);
```

**Review by willeastcott — APPROVED** [review]

File: src/framework/parsers/texture/img.js (hunk 4 above)
Comment (mvaligursky) [reply src/framework/parsers/texture/img.js]: I should! ta.

**Pull request merged**